File: solutions/java/src/restaurantmanagementsystem/command/ServeOrderCommand.java
package restaurantmanagementsystem.command;

import restaurantmanagementsystem.model.Order;
import restaurantmanagementsystem.model.Waiter;

public class ServeOrderCommand implements Command{
    private final Order order;
    private final Waiter waiter;

    public ServeOrderCommand(Order order, Waiter waiter) {
        this.order = order;
        this.waiter = waiter;
    }

    @Override
    public void execute() {
        waiter.serveOrder(order);
    }
}

File: solutions/csharp/restaurantmanagementsystem/Commands/ServeOrderCommand.cs
class ServeOrderCommand : ICommand
{
    private readonly Order order;
    private readonly Waiter waiter;

    public ServeOrderCommand(Order order, Waiter waiter)
    {
        this.order = order;
        this.waiter = waiter;
    }

    public void Execute()
    {
        waiter.ServeOrder(order);
    }
}